pkg/prometheus/statefulset_test.go

```diff
@@ -650,6 +650,75 @@ func TestTagAndShaAndVersion(t *testing.T) {
 	}
 }
 
+func TestPrometheusDefaultBaseImageFlag(t *testing.T) {
```

Comment:
I'd like to see more tests to cover the various combinations. [Table-driven tests](https://github.com/golang/go/wiki/TableDrivenTests) would reduce duplication. I have [something](https://github.com/simonpasquier/prometheus-operator/blob/6d4136af478965acb85cb7bf6f3b499bb9986dc7/pkg/alertmanager/statefulset_test.go#L450) in a branch of my personal fork (I started to work on the fix but I didn't have time to finish), feel free to have a look and get inspiration.

Reply:
Sure. I will have a look and try to put a separate PR.

Reply:
Prometheus:
https://github.com/prometheus-operator/prometheus-operator/blob/30b03e839a460ef230256873c10ad56f9a5abd37/pkg/prometheus/statefulset.go#L320-L324

Additional Thanos in Prometheues:
https://github.com/prometheus-operator/prometheus-operator/blob/30b03e839a460ef230256873c10ad56f9a5abd37/pkg/prometheus/statefulset.go#L686-L698

Alertmanager:
https://github.com/prometheus-operator/prometheus-operator/blob/30b03e839a460ef230256873c10ad56f9a5abd37/pkg/alertmanager/statefulset.go#L213-L223

Thanos:
https://github.com/prometheus-operator/prometheus-operator/blob/30b03e839a460ef230256873c10ad56f9a5abd37/pkg/thanos/statefulset.go#L155-L162

Which pattern above is to be used across all?
SHA and Tag is not available for the Thanos - not sure why?